File: 20190330_Saturday/20190330.js
console.log("alias x='node 20190330_Saturday/20190330.js'");
// Generate: 36 sectonds; Sort: 110 seconds

var startTime, endTime;

function start() {
  startTime = new Date();
};

function end() {
  endTime = new Date();
  var timeDiff = endTime - startTime; //in ms
  // strip the ms
  timeDiff /= 1000;

  // get seconds 
  var seconds = Math.round(timeDiff);
  console.log(seconds + " seconds");
}

function shuffle(a) {
    for (let i = a.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [a[i], a[j]] = [a[j], a[i]];
    }
    return a;
}

let mln_arr = () => shuffle(Array.from({length: 1000000}, (_,i) => i));
let buffer = []
console.log("Generate arrays:");
start();
for(i=0; i<=100; i++) buffer.push(mln_arr());
console.log(end());

console.log("Sort arrays:")
start();
buffer.forEach(x => x.sort())
console.log(end());

File: 20190330_Saturday/20190330.lua
print("alias x='lua 20190330_Saturday/20190330.lua'")
-- isert: 75sec, sort: 153 sec
function shuffle(tbl)
    size = #tbl
    for i = size, 1, -1 do
      local rand = math.random(size)
      tbl[i], tbl[rand] = tbl[rand], tbl[i]
    end
    return tbl
  end

  function arr()
    tbl = {}
    for i = 1,1000000,1 do
        table.insert(tbl,i)
    end
    return tbl
  end

print("insert")
local x = os.clock()
buffer = {}
for i=1,100 do
    table.insert(buffer,shuffle(arr()))
end
print(string.format("Insert time: %.2f\n", os.clock() - x))

 x = os.clock()
print("sort")
for i=1,100 do
    table.sort(buffer[i])
end
print(string.format("Sort time: %.2f\n", os.clock() - x))
